servicetest/networkgateway/test_network_gateway.py

```diff
@@ -187,6 +203,53 @@ def test_allow_ping_only_for_specific_domain(self):
         finally:
             sc.terminate()
 
+    def test_whitelist(self):
+        """ Tests the whitelisting feauture.
+            a config for allowing ping to example.com is set first
+            then another config for allowing ping to iana.org is set
+            consequences are questioned with asserts
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+
+            sc.set_gateway_config("network", GW_CONFIG_ACCEPT_EXAMPLE_COM)
+            sc.set_gateway_config("network", GW_CONFIG_ACCEPT_IANA_ORG)
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py" +
+                              " --test-dir " + sc.get_bind_dir() +
+                              " --do-ping iana.org")
+            time.sleep(2)
+            helper = NetworkHelper(CURRENT_DIR)
+            is_pingable = helper.ping_result()
+            assert is_pingable is True
+            helper.remove_file()
+
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py" +
+                              " --test-dir " + sc.get_bind_dir() +
+                              " --do-ping example.org")
+            time.sleep(2)
+            helper = NetworkHelper(CURRENT_DIR)
+            is_pingable = helper.ping_result()
+            assert is_pingable is True
+            helper.remove_file()
+
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py" +
+                              " --test-dir " + sc.get_bind_dir() +
+                              " --do-ping pelagicore.com")
+            time.sleep(10)
```

Comment:
Could you add some comments describing when we do long-ish sleeps? I think it should be motivated so anyone maintaining this doesn't need to think too much about it.